core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -77,6 +77,18 @@ class StorageStatusListener(conf: SparkConf) extends SparkListener {
       val maxMem = blockManagerAdded.maxMem
       val storageStatus = new StorageStatus(blockManagerId, maxMem)
       executorIdToStorageStatus(executorId) = storageStatus
+
+      // Try to remove the dead storage status if same executor register the block manger twice.
+      removeDeadExecutorStorageStatus(executorId)
+    }
+  }
+
+  private def removeDeadExecutorStorageStatus(executorId: String): Unit = {
+    deadExecutorStorageStatus.zipWithIndex.foreach { case (status, index) =>
```

Comment:
Ah. OK what about 

```
val toRemove = deadExecutorStorageStatus.find(_.blockManagerId.executorId == executorId)
if (toRemove >= 0) {
  deadExecutorStorageStatus.remove(toRemove)
}
```
Might be clearer and more direct. No big deal either way.